test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs

```diff
@@ -180,6 +180,43 @@ public void JWETouchAllProperties()
             TestUtilities.AssertFailIfErrors(context);
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void CheckAuthenticationTagAndCiphertextValues(bool isEncrypted)
+        {
+            var jsonWebTokenHandler = new JsonWebTokenHandler();
+            SecurityTokenDescriptor tokenDescriptor = new SecurityTokenDescriptor
+            {
+                SigningCredentials = KeyingMaterial.JsonWebKeyRsa256SigningCredentials,
+                Subject = new ClaimsIdentity(Default.PayloadClaims),
+                TokenType = "TokenType"
+            };
+
+            if (isEncrypted)
+            {
+                tokenDescriptor.EncryptingCredentials = KeyingMaterial.DefaultSymmetricEncryptingCreds_Aes256_Sha512_512;
+            }
+
+            string jwt = jsonWebTokenHandler.CreateToken(tokenDescriptor);
+            JsonWebToken jsonWebToken = new JsonWebToken(jwt);
+
+            if (isEncrypted)
+            {
+                Assert.NotNull(jsonWebToken.AuthenticationTag);
+                Assert.NotNull(jsonWebToken.AuthenticationTagBytes);
+                Assert.NotNull(jsonWebToken.Ciphertext);
+                Assert.NotNull(jsonWebToken.CipherTextBytes);
+            }
+            else
+            {
+                Assert.Empty(jsonWebToken.AuthenticationTag);
```

Comment:
Should we apply the same thing for AuthenticationTag? Currently `string.Empty` is returning if the token is not encrypted.